components/sidebar/SearchView/SearchView.tsx

```diff
@@ -34,20 +34,25 @@ import SearchViewHints from "./SearchViewHints"
 const MotionFlex = motion<FlexProps>(Flex)
 const MotionButton = motion<ButtonProps>(Button)
 
+interface SearchViewProps {
+  searchQuery: string
+  setSearchQuery: Dispatch<React.SetStateAction<string>>
+  searchOffset: number
+  setSearchOffset: Dispatch<React.SetStateAction<number>>
+  chosenCourse: string
+  setChosenCourse: Dispatch<React.SetStateAction<string>>
+}
+
 const SearchView = ({
   searchQuery,
   setSearchQuery,
   searchOffset,
   setSearchOffset,
-}: {
-  searchQuery: string
-  setSearchQuery: Dispatch<React.SetStateAction<string>>
-  searchOffset: number
-  setSearchOffset: Dispatch<React.SetStateAction<number>>
-}) => {
+  chosenCourse,
```

Comment:
If I remember correctly, the sidebar course is in the app context? Can you give a tldr of this new structure?